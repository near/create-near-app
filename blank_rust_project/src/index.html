<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8">
  <link rel="stylesheet" href="https://stackpath.bootstrapcdn.com/bootstrap/4.2.1/css/bootstrap.min.css" integrity="sha384-GJzZqFGwb1QTTN6wy59ffF1BuGJpLSa9DkKMp0DgiMDm4iYMj70gZWKYbI706tWS" crossorigin="anonymous">
  <style>
    @font-face {
      font-family: BwSeidoRound;
      src: url(./fonts/BwSeidoRound500.eot);
      src: url(./fonts/BwSeidoRound500.eot?#iefix) format("embedded-opentype"),
      url(./fonts/BwSeidoRound500.woff2) format("woff2"),
      url(./fonts/BwSeidoRound500.woff) format("woff"),
      url(./fonts/BwSeidoRound500.ttf) format("truetype");
      font-weight: 500;
      font-style: normal;
    }

    .App-header {
      background-image: url('https://near.org/wp-content/uploads/2019/03/illo-developers.svg');
      min-height: 100vh;
      display: flex;
      flex-direction: column;
      align-items: center;
      justify-content: center;
      font-size: 1.3rem;
      color: #25282A;
      text-align: center;
      padding: 2%;
    }

    .image-wrapper {
      display: flex;
      align-items: center;
      flex-direction: column;
      justify-content: center;
    }

    .App-logo {
      animation: App-logo-spin infinite 20s linear;
      pointer-events: none;
    }

    .margin-logo {
      margin: 20px;
    }

    p, button, a {
      font-family: BwSeidoRound;
    }

    a:hover {
      color: #0072CE;
    }

    a:visited {
      color: #8FD6BD;
    }

    code {
      font-size: 1.5rem;
    }

    @keyframes App-logo-spin {
      from {
          transform: rotate(0deg);
      }
      to {
          transform: rotate(360deg);
      }
    }

    button {
      padding: 5px;
      color: #0072CE;
      border-radius: 10px;
      border: 3px solid #0072CE;
      margin: 0 10px;
      font-size: 1.1rem;
    }

    button:hover {
      cursor: pointer;
      color: white;
      background: #0072CE;
    }

    .speech {
      font-size: 1.5rem;
      color: #0072CE;
      text-shadow: 1px 1px #D1CCBD;
    }
  </style>
</head>
<body class="App-header">
  <div>
      <div class="image-wrapper">
        <img style="width:33%;" src="assets/gray_near_logo.svg" alt="NEAR logo" />
        <p><span role="img" aria-label="fish">üêü</span> NEAR protocol is a new blockchain focused on developer productivity and useability!<span role="img" aria-label="fish">üêü</span></p>
        <p><span role="img" aria-label="chain">‚õì</span> This little app is connected to blockchain right now. <span role="img" aria-label="chain">‚õì</span></p>
        <img src="assets/near.svg" class="App-logo margin-logo" alt="logo" />
        <p id="speech" class="speech"></p>
      </div>
      <div id="signed-out-flow" class="d-none">
        <button id="sign-in-button" >Sign-in with NEAR</button>
      </div>
      <div id="signed-in-flow" class="d-none">
          <button id="sign-out-button" >Sign-out</button>
          <button id="change-greeting">Change greeting</button>
      </div> 
  </div>
  <script src="./main.js"></script>
</body>
</html>
