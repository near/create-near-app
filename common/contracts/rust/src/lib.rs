/*
 * This is an example of a Rust smart contract with two simple, symmetric functions:
 *
 * 1. set_greeting: accepts a greeting, such as "howdy", and records it for the user (account_id)
 *    who sent the request
 * 2. get_greeting: accepts an account_id and returns the greeting saved for it, defaulting to
 *    "Hello"
 *
 * Learn more about writing NEAR smart contracts with Rust:
 * https://github.com/near/near-sdk-rs
 *
 */

// To conserve gas, efficient serialization is achieved through Borsh (http://borsh.io/)
use near_sdk::borsh::{self, BorshDeserialize, BorshSerialize};
use near_sdk::{env, near_bindgen, setup_alloc};
use near_sdk::collections::LookupMap;

setup_alloc!();

// Structs in Rust are similar to other languages, and may include impl keyword as shown below
// Note: the names of the structs are not important when calling the smart contract, but the function names are
#[near_bindgen]
#[derive(BorshDeserialize, BorshSerialize)]
pub struct Welcome {
    records: LookupMap<String, String>,
}

impl Default for Welcome {
  fn default() -> Self {
    Self {
      records: LookupMap::new(b"a".to_vec()),
    }
  }
}

#[near_bindgen]
impl Welcome {
    pub fn set_greeting(&mut self, message: String) {
        let account_id = env::signer_account_id();

        // Use env::log to record logs permanently to the blockchain!
        env::log(format!("Saving greeting '{}' for account '{}'", message, account_id,).as_bytes());

        self.records.insert(&account_id, &message);
    }

    // `match` is similar to `switch` in other languages; here we use it to default to "Hello" if
    // self.records.get(&account_id) is not yet defined.
    // Learn more: https://doc.rust-lang.org/book/ch06-02-match.html#matching-with-optiont
    pub fn get_greeting(&self, account_id: String) -> String {
        match self.records.get(&account_id) {
            Some(greeting) => greeting,
            None => "Hello".to_string(),
        }
    }
}

/*
 * The rest of this file holds the inline tests for the code above
 * Learn more about Rust tests: https://doc.rust-lang.org/book/ch11-01-writing-tests.html
 *
 * To run from contract directory:
 * cargo test -- --nocapture
 *
 * From project root, to run in combination with frontend tests:
 * yarn test
 *
 */
#[cfg(test)]
mod tests {
    use super::*;
    use near_sdk::MockedBlockchain;
    use near_sdk::{testing_env, VMContext};

    // mock the context for testing, notice "signer_account_id" that was accessed above from env::
    fn get_context(input: Vec<u8>, is_view: bool) -> VMContext {
        VMContext {
            current_account_id: "alice_near".to_string(),
            signer_account_id: "bob_near".to_string(),
            signer_account_pk: vec![0, 1, 2],
            predecessor_account_id: "carol_near".to_string(),
            input,
            block_index: 0,
            block_timestamp: 0,
            account_balance: 0,
            account_locked_balance: 0,
            storage_usage: 0,
            attached_deposit: 0,
            prepaid_gas: 10u64.pow(18),
            random_seed: vec![0, 1, 2],
            is_view,
            output_data_receivers: vec![],
            epoch_height: 19,
        }
    }

    #[test]
    fn set_then_get_greeting() {
        let context = get_context(vec![], false);
        testing_env!(context);
        let mut contract = Welcome::default();
        contract.set_greeting("howdy".to_string());
        assert_eq!(
            "howdy".to_string(),
            contract.get_greeting("bob_near".to_string())
        );
    }

    #[test]
    fn get_default_greeting() {
        let context = get_context(vec![], true);
        testing_env!(context);
        let contract = Welcome::default();
        // this test did not call set_greeting so should return the default "Hello" greeting
        assert_eq!(
            "Hello".to_string(),
            contract.get_greeting("francis.near".to_string())
        );
    }
}
