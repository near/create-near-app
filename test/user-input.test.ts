import * as show from '../src/messages';
import SpyInstance = jest.SpyInstance;

describe('messages', () => {
  let showSpy: SpyInstance;
  test('snapshot basic messages', () => {
    showSpy = jest.spyOn(show, 'show').mockImplementation(() => {});
    show.welcome();
    show.setupFailed();
    show.argsError('wrong args');
    show.windowsWarning();
    show.creatingApp();
    show.depsInstall();
    show.depsInstallError();
    expect(showSpy.mock.calls).toMatchSnapshot();
    showSpy.mockClear();
  });

  test('snapshot messages with params', () => {
    showSpy = jest.spyOn(show, 'show').mockImplementation(() => {});

    show.successContractToText('ts');
    show.successContractToText('rs');

    show.successFrontendToText('next-page');
    show.successFrontendToText('next-app');

    show.unsupportedNodeVersion('55.789');
    show.directoryExists('/a/b/c/d');

    expect(showSpy.mock.calls).toMatchSnapshot();
    showSpy.mockClear();
  });

});