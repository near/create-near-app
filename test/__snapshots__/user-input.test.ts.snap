// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`messages snapshot basic messages 1`] = `
[
  [
    "[34m======================================================[39m
üëã [1m[32mWelcome to NEAR![39m[22m Learn more: https://docs.near.org/
üîß Let's get your dApp ready.
[34m======================================================[39m
(NEAR collects anonymous information on the commands used. No personal information that could identify you is shared)
",
  ],
  [
    "[1m[31m==========================================[39m[22m
[31m‚õîÔ∏è There was a problem during NEAR project setup[39m.
Please refer to https://github.com/near/create-near-app README to troubleshoot.
Notice: some platforms aren't supported (yet).
[1m[31m==========================================[39m[22m",
  ],
  [
    "[31mArguments error[39m
Run [34mnpx create-near-app[39m without arguments, or use:
npx create-near-app <projectName> --contract rust|js --frontend react|vanilla|none --tests js|rust",
  ],
  [
    "[43m[30mNotice: On Win32 please use WSL (Windows Subsystem for Linux).[39m[49m
https://docs.microsoft.com/en-us/windows/wsl/install
Exiting now.",
  ],
  [
    "
Creating a new [1mNEAR dApp[22m",
  ],
  [
    "
[32mInstalling dependencies in a few folders, this might take a while.[39m
",
  ],
  [
    "[31mError installing NEAR project dependencies[39m",
  ],
]
`;

exports[`messages snapshot messages with params 1`] = `
[
  [
    "[31mWe support node.js version 55.789 or later[39m",
  ],
  [
    "[31mThis directory already exists! /a/b/c/d[39m",
  ],
]
`;

exports[`test success message test matrix 'js' 'none' 'js' false 1`] = `
[
  [
    "
[32m======================================================[39m
‚úÖ  Success! Created 'my_project_name'
   with a smart contract in [1mJavaScript[22m.

  [1m[43m[30mYour next steps[39m[49m[22m:
   - [7mNavigate to your project[27m:
         [34mcd [39m[34m[1mmy_project_name[22m[39m
   - [7mInstall all dependencies[27m
         [34mnpm [39m[34m[1minstall[22m[39m
   - [7mBuild your contract[27m:
         [34mnpm [39m[34m[1mrun build[22m[39m
   - [7mTest your contract[27m in NEAR SandBox:
         [34mnpm [39m[34m[1mtest[22m[39m
   - [7mDeploy your contract[27m to NEAR TestNet with a temporary dev account:
         [34mnpm [39m[34m[1mrun deploy[22m[39m
   
üß† Read [1m[92mREADME.md[39m[22m to explore further.",
  ],
]
`;

exports[`test success message test matrix 'js' 'none' 'js' true 1`] = `
[
  [
    "
[32m======================================================[39m
‚úÖ  Success! Created 'my_project_name'
   with a smart contract in [1mJavaScript[22m.

  [1m[43m[30mYour next steps[39m[49m[22m:
   - [7mNavigate to your project[27m:
         [34mcd [39m[34m[1mmy_project_name[22m[39m
   Then:
   - [7mBuild your contract[27m:
         [34mnpm [39m[34m[1mrun build[22m[39m
   - [7mTest your contract[27m in NEAR SandBox:
         [34mnpm [39m[34m[1mtest[22m[39m
   - [7mDeploy your contract[27m to NEAR TestNet with a temporary dev account:
         [34mnpm [39m[34m[1mrun deploy[22m[39m
   
üß† Read [1m[92mREADME.md[39m[22m to explore further.",
  ],
]
`;

exports[`test success message test matrix 'js' 'none' 'rust' false 1`] = `
[
  [
    "
[32m======================================================[39m
‚úÖ  Success! Created 'my_project_name'
   with a smart contract in [1mJavaScript[22m.

  [1m[43m[30mYour next steps[39m[49m[22m:
   - [7mNavigate to your project[27m:
         [34mcd [39m[34m[1mmy_project_name[22m[39m
   - [7mInstall all dependencies[27m
         [34mnpm [39m[34m[1minstall[22m[39m
   - [7mBuild your contract[27m:
         [34mnpm [39m[34m[1mrun build[22m[39m
   - [7mTest your contract[27m in NEAR SandBox:
         [34mnpm [39m[34m[1mtest[22m[39m
   - [7mDeploy your contract[27m to NEAR TestNet with a temporary dev account:
         [34mnpm [39m[34m[1mrun deploy[22m[39m
   
üß† Read [1m[92mREADME.md[39m[22m to explore further.",
  ],
]
`;

exports[`test success message test matrix 'js' 'none' 'rust' true 1`] = `
[
  [
    "
[32m======================================================[39m
‚úÖ  Success! Created 'my_project_name'
   with a smart contract in [1mJavaScript[22m.

  [1m[43m[30mYour next steps[39m[49m[22m:
   - [7mNavigate to your project[27m:
         [34mcd [39m[34m[1mmy_project_name[22m[39m
   Then:
   - [7mBuild your contract[27m:
         [34mnpm [39m[34m[1mrun build[22m[39m
   - [7mTest your contract[27m in NEAR SandBox:
         [34mnpm [39m[34m[1mtest[22m[39m
   - [7mDeploy your contract[27m to NEAR TestNet with a temporary dev account:
         [34mnpm [39m[34m[1mrun deploy[22m[39m
   
üß† Read [1m[92mREADME.md[39m[22m to explore further.",
  ],
]
`;

exports[`test success message test matrix 'js' 'react' 'js' false 1`] = `
[
  [
    "
[32m======================================================[39m
‚úÖ  Success! Created 'my_project_name'
   with a smart contract in [1mJavaScript[22m and a frontend template[1m in React.js[22m.

  [1m[43m[30mYour next steps[39m[49m[22m:
   - [7mNavigate to your project[27m:
         [34mcd [39m[34m[1mmy_project_name[22m[39m
   - [7mInstall all dependencies[27m
         [34mnpm [39m[34m[1minstall[22m[39m
   - [7mBuild your contract[27m:
         [34mnpm [39m[34m[1mrun build[22m[39m
   - [7mTest your contract[27m in NEAR SandBox:
         [34mnpm [39m[34m[1mtest[22m[39m
   - [7mDeploy your contract[27m to NEAR TestNet with a temporary dev account:
         [34mnpm [39m[34m[1mrun deploy[22m[39m
   - [7mStart your frontend[27m:
         [34mnpm [39m[34m[1mstart[22m[39m

üß† Read [1m[92mREADME.md[39m[22m to explore further.",
  ],
]
`;

exports[`test success message test matrix 'js' 'react' 'js' true 1`] = `
[
  [
    "
[32m======================================================[39m
‚úÖ  Success! Created 'my_project_name'
   with a smart contract in [1mJavaScript[22m and a frontend template[1m in React.js[22m.

  [1m[43m[30mYour next steps[39m[49m[22m:
   - [7mNavigate to your project[27m:
         [34mcd [39m[34m[1mmy_project_name[22m[39m
   Then:
   - [7mBuild your contract[27m:
         [34mnpm [39m[34m[1mrun build[22m[39m
   - [7mTest your contract[27m in NEAR SandBox:
         [34mnpm [39m[34m[1mtest[22m[39m
   - [7mDeploy your contract[27m to NEAR TestNet with a temporary dev account:
         [34mnpm [39m[34m[1mrun deploy[22m[39m
   - [7mStart your frontend[27m:
         [34mnpm [39m[34m[1mstart[22m[39m

üß† Read [1m[92mREADME.md[39m[22m to explore further.",
  ],
]
`;

exports[`test success message test matrix 'js' 'react' 'rust' false 1`] = `
[
  [
    "
[32m======================================================[39m
‚úÖ  Success! Created 'my_project_name'
   with a smart contract in [1mJavaScript[22m and a frontend template[1m in React.js[22m.

  [1m[43m[30mYour next steps[39m[49m[22m:
   - [7mNavigate to your project[27m:
         [34mcd [39m[34m[1mmy_project_name[22m[39m
   - [7mInstall all dependencies[27m
         [34mnpm [39m[34m[1minstall[22m[39m
   - [7mBuild your contract[27m:
         [34mnpm [39m[34m[1mrun build[22m[39m
   - [7mTest your contract[27m in NEAR SandBox:
         [34mnpm [39m[34m[1mtest[22m[39m
   - [7mDeploy your contract[27m to NEAR TestNet with a temporary dev account:
         [34mnpm [39m[34m[1mrun deploy[22m[39m
   - [7mStart your frontend[27m:
         [34mnpm [39m[34m[1mstart[22m[39m

üß† Read [1m[92mREADME.md[39m[22m to explore further.",
  ],
]
`;

exports[`test success message test matrix 'js' 'react' 'rust' true 1`] = `
[
  [
    "
[32m======================================================[39m
‚úÖ  Success! Created 'my_project_name'
   with a smart contract in [1mJavaScript[22m and a frontend template[1m in React.js[22m.

  [1m[43m[30mYour next steps[39m[49m[22m:
   - [7mNavigate to your project[27m:
         [34mcd [39m[34m[1mmy_project_name[22m[39m
   Then:
   - [7mBuild your contract[27m:
         [34mnpm [39m[34m[1mrun build[22m[39m
   - [7mTest your contract[27m in NEAR SandBox:
         [34mnpm [39m[34m[1mtest[22m[39m
   - [7mDeploy your contract[27m to NEAR TestNet with a temporary dev account:
         [34mnpm [39m[34m[1mrun deploy[22m[39m
   - [7mStart your frontend[27m:
         [34mnpm [39m[34m[1mstart[22m[39m

üß† Read [1m[92mREADME.md[39m[22m to explore further.",
  ],
]
`;

exports[`test success message test matrix 'js' 'vanilla' 'js' false 1`] = `
[
  [
    "
[32m======================================================[39m
‚úÖ  Success! Created 'my_project_name'
   with a smart contract in [1mJavaScript[22m and a frontend template.

  [1m[43m[30mYour next steps[39m[49m[22m:
   - [7mNavigate to your project[27m:
         [34mcd [39m[34m[1mmy_project_name[22m[39m
   - [7mInstall all dependencies[27m
         [34mnpm [39m[34m[1minstall[22m[39m
   - [7mBuild your contract[27m:
         [34mnpm [39m[34m[1mrun build[22m[39m
   - [7mTest your contract[27m in NEAR SandBox:
         [34mnpm [39m[34m[1mtest[22m[39m
   - [7mDeploy your contract[27m to NEAR TestNet with a temporary dev account:
         [34mnpm [39m[34m[1mrun deploy[22m[39m
   - [7mStart your frontend[27m:
         [34mnpm [39m[34m[1mstart[22m[39m

üß† Read [1m[92mREADME.md[39m[22m to explore further.",
  ],
]
`;

exports[`test success message test matrix 'js' 'vanilla' 'js' true 1`] = `
[
  [
    "
[32m======================================================[39m
‚úÖ  Success! Created 'my_project_name'
   with a smart contract in [1mJavaScript[22m and a frontend template.

  [1m[43m[30mYour next steps[39m[49m[22m:
   - [7mNavigate to your project[27m:
         [34mcd [39m[34m[1mmy_project_name[22m[39m
   Then:
   - [7mBuild your contract[27m:
         [34mnpm [39m[34m[1mrun build[22m[39m
   - [7mTest your contract[27m in NEAR SandBox:
         [34mnpm [39m[34m[1mtest[22m[39m
   - [7mDeploy your contract[27m to NEAR TestNet with a temporary dev account:
         [34mnpm [39m[34m[1mrun deploy[22m[39m
   - [7mStart your frontend[27m:
         [34mnpm [39m[34m[1mstart[22m[39m

üß† Read [1m[92mREADME.md[39m[22m to explore further.",
  ],
]
`;

exports[`test success message test matrix 'js' 'vanilla' 'rust' false 1`] = `
[
  [
    "
[32m======================================================[39m
‚úÖ  Success! Created 'my_project_name'
   with a smart contract in [1mJavaScript[22m and a frontend template.

  [1m[43m[30mYour next steps[39m[49m[22m:
   - [7mNavigate to your project[27m:
         [34mcd [39m[34m[1mmy_project_name[22m[39m
   - [7mInstall all dependencies[27m
         [34mnpm [39m[34m[1minstall[22m[39m
   - [7mBuild your contract[27m:
         [34mnpm [39m[34m[1mrun build[22m[39m
   - [7mTest your contract[27m in NEAR SandBox:
         [34mnpm [39m[34m[1mtest[22m[39m
   - [7mDeploy your contract[27m to NEAR TestNet with a temporary dev account:
         [34mnpm [39m[34m[1mrun deploy[22m[39m
   - [7mStart your frontend[27m:
         [34mnpm [39m[34m[1mstart[22m[39m

üß† Read [1m[92mREADME.md[39m[22m to explore further.",
  ],
]
`;

exports[`test success message test matrix 'js' 'vanilla' 'rust' true 1`] = `
[
  [
    "
[32m======================================================[39m
‚úÖ  Success! Created 'my_project_name'
   with a smart contract in [1mJavaScript[22m and a frontend template.

  [1m[43m[30mYour next steps[39m[49m[22m:
   - [7mNavigate to your project[27m:
         [34mcd [39m[34m[1mmy_project_name[22m[39m
   Then:
   - [7mBuild your contract[27m:
         [34mnpm [39m[34m[1mrun build[22m[39m
   - [7mTest your contract[27m in NEAR SandBox:
         [34mnpm [39m[34m[1mtest[22m[39m
   - [7mDeploy your contract[27m to NEAR TestNet with a temporary dev account:
         [34mnpm [39m[34m[1mrun deploy[22m[39m
   - [7mStart your frontend[27m:
         [34mnpm [39m[34m[1mstart[22m[39m

üß† Read [1m[92mREADME.md[39m[22m to explore further.",
  ],
]
`;

exports[`test success message test matrix 'rust' 'none' 'js' false 1`] = `
[
  [
    "
[32m======================================================[39m
‚úÖ  Success! Created 'my_project_name'
   with a smart contract in [1mRust[22m.
ü¶Ä If you are new to Rust please visit [1m[32mhttps://www.rust-lang.org [39m[22m

  [1m[43m[30mYour next steps[39m[49m[22m:
   - [7mNavigate to your project[27m:
         [34mcd [39m[34m[1mmy_project_name[22m[39m
   - [7mInstall all dependencies[27m
         [34mnpm [39m[34m[1minstall[22m[39m
   - [7mBuild your contract[27m:
         [34mnpm [39m[34m[1mrun build[22m[39m
   - [7mTest your contract[27m in NEAR SandBox:
         [34mnpm [39m[34m[1mtest[22m[39m
   - [7mDeploy your contract[27m to NEAR TestNet with a temporary dev account:
         [34mnpm [39m[34m[1mrun deploy[22m[39m
   
üß† Read [1m[92mREADME.md[39m[22m to explore further.",
  ],
]
`;

exports[`test success message test matrix 'rust' 'none' 'js' true 1`] = `
[
  [
    "
[32m======================================================[39m
‚úÖ  Success! Created 'my_project_name'
   with a smart contract in [1mRust[22m.
ü¶Ä If you are new to Rust please visit [1m[32mhttps://www.rust-lang.org [39m[22m

  [1m[43m[30mYour next steps[39m[49m[22m:
   - [7mNavigate to your project[27m:
         [34mcd [39m[34m[1mmy_project_name[22m[39m
   Then:
   - [7mBuild your contract[27m:
         [34mnpm [39m[34m[1mrun build[22m[39m
   - [7mTest your contract[27m in NEAR SandBox:
         [34mnpm [39m[34m[1mtest[22m[39m
   - [7mDeploy your contract[27m to NEAR TestNet with a temporary dev account:
         [34mnpm [39m[34m[1mrun deploy[22m[39m
   
üß† Read [1m[92mREADME.md[39m[22m to explore further.",
  ],
]
`;

exports[`test success message test matrix 'rust' 'none' 'rust' false 1`] = `
[
  [
    "
[32m======================================================[39m
‚úÖ  Success! Created 'my_project_name'
   with a smart contract in [1mRust[22m.
ü¶Ä If you are new to Rust please visit [1m[32mhttps://www.rust-lang.org [39m[22m

  [1m[43m[30mYour next steps[39m[49m[22m:
   - [7mNavigate to your project[27m:
         [34mcd [39m[34m[1mmy_project_name[22m[39m
   - [7mInstall all dependencies[27m
         [34mnpm [39m[34m[1minstall[22m[39m
   - [7mBuild your contract[27m:
         [34mnpm [39m[34m[1mrun build[22m[39m
   - [7mTest your contract[27m in NEAR SandBox:
         [34mnpm [39m[34m[1mtest[22m[39m
   - [7mDeploy your contract[27m to NEAR TestNet with a temporary dev account:
         [34mnpm [39m[34m[1mrun deploy[22m[39m
   
üß† Read [1m[92mREADME.md[39m[22m to explore further.",
  ],
]
`;

exports[`test success message test matrix 'rust' 'none' 'rust' true 1`] = `
[
  [
    "
[32m======================================================[39m
‚úÖ  Success! Created 'my_project_name'
   with a smart contract in [1mRust[22m.
ü¶Ä If you are new to Rust please visit [1m[32mhttps://www.rust-lang.org [39m[22m

  [1m[43m[30mYour next steps[39m[49m[22m:
   - [7mNavigate to your project[27m:
         [34mcd [39m[34m[1mmy_project_name[22m[39m
   Then:
   - [7mBuild your contract[27m:
         [34mnpm [39m[34m[1mrun build[22m[39m
   - [7mTest your contract[27m in NEAR SandBox:
         [34mnpm [39m[34m[1mtest[22m[39m
   - [7mDeploy your contract[27m to NEAR TestNet with a temporary dev account:
         [34mnpm [39m[34m[1mrun deploy[22m[39m
   
üß† Read [1m[92mREADME.md[39m[22m to explore further.",
  ],
]
`;

exports[`test success message test matrix 'rust' 'react' 'js' false 1`] = `
[
  [
    "
[32m======================================================[39m
‚úÖ  Success! Created 'my_project_name'
   with a smart contract in [1mRust[22m and a frontend template[1m in React.js[22m.
ü¶Ä If you are new to Rust please visit [1m[32mhttps://www.rust-lang.org [39m[22m

  [1m[43m[30mYour next steps[39m[49m[22m:
   - [7mNavigate to your project[27m:
         [34mcd [39m[34m[1mmy_project_name[22m[39m
   - [7mInstall all dependencies[27m
         [34mnpm [39m[34m[1minstall[22m[39m
   - [7mBuild your contract[27m:
         [34mnpm [39m[34m[1mrun build[22m[39m
   - [7mTest your contract[27m in NEAR SandBox:
         [34mnpm [39m[34m[1mtest[22m[39m
   - [7mDeploy your contract[27m to NEAR TestNet with a temporary dev account:
         [34mnpm [39m[34m[1mrun deploy[22m[39m
   - [7mStart your frontend[27m:
         [34mnpm [39m[34m[1mstart[22m[39m

üß† Read [1m[92mREADME.md[39m[22m to explore further.",
  ],
]
`;

exports[`test success message test matrix 'rust' 'react' 'js' true 1`] = `
[
  [
    "
[32m======================================================[39m
‚úÖ  Success! Created 'my_project_name'
   with a smart contract in [1mRust[22m and a frontend template[1m in React.js[22m.
ü¶Ä If you are new to Rust please visit [1m[32mhttps://www.rust-lang.org [39m[22m

  [1m[43m[30mYour next steps[39m[49m[22m:
   - [7mNavigate to your project[27m:
         [34mcd [39m[34m[1mmy_project_name[22m[39m
   Then:
   - [7mBuild your contract[27m:
         [34mnpm [39m[34m[1mrun build[22m[39m
   - [7mTest your contract[27m in NEAR SandBox:
         [34mnpm [39m[34m[1mtest[22m[39m
   - [7mDeploy your contract[27m to NEAR TestNet with a temporary dev account:
         [34mnpm [39m[34m[1mrun deploy[22m[39m
   - [7mStart your frontend[27m:
         [34mnpm [39m[34m[1mstart[22m[39m

üß† Read [1m[92mREADME.md[39m[22m to explore further.",
  ],
]
`;

exports[`test success message test matrix 'rust' 'react' 'rust' false 1`] = `
[
  [
    "
[32m======================================================[39m
‚úÖ  Success! Created 'my_project_name'
   with a smart contract in [1mRust[22m and a frontend template[1m in React.js[22m.
ü¶Ä If you are new to Rust please visit [1m[32mhttps://www.rust-lang.org [39m[22m

  [1m[43m[30mYour next steps[39m[49m[22m:
   - [7mNavigate to your project[27m:
         [34mcd [39m[34m[1mmy_project_name[22m[39m
   - [7mInstall all dependencies[27m
         [34mnpm [39m[34m[1minstall[22m[39m
   - [7mBuild your contract[27m:
         [34mnpm [39m[34m[1mrun build[22m[39m
   - [7mTest your contract[27m in NEAR SandBox:
         [34mnpm [39m[34m[1mtest[22m[39m
   - [7mDeploy your contract[27m to NEAR TestNet with a temporary dev account:
         [34mnpm [39m[34m[1mrun deploy[22m[39m
   - [7mStart your frontend[27m:
         [34mnpm [39m[34m[1mstart[22m[39m

üß† Read [1m[92mREADME.md[39m[22m to explore further.",
  ],
]
`;

exports[`test success message test matrix 'rust' 'react' 'rust' true 1`] = `
[
  [
    "
[32m======================================================[39m
‚úÖ  Success! Created 'my_project_name'
   with a smart contract in [1mRust[22m and a frontend template[1m in React.js[22m.
ü¶Ä If you are new to Rust please visit [1m[32mhttps://www.rust-lang.org [39m[22m

  [1m[43m[30mYour next steps[39m[49m[22m:
   - [7mNavigate to your project[27m:
         [34mcd [39m[34m[1mmy_project_name[22m[39m
   Then:
   - [7mBuild your contract[27m:
         [34mnpm [39m[34m[1mrun build[22m[39m
   - [7mTest your contract[27m in NEAR SandBox:
         [34mnpm [39m[34m[1mtest[22m[39m
   - [7mDeploy your contract[27m to NEAR TestNet with a temporary dev account:
         [34mnpm [39m[34m[1mrun deploy[22m[39m
   - [7mStart your frontend[27m:
         [34mnpm [39m[34m[1mstart[22m[39m

üß† Read [1m[92mREADME.md[39m[22m to explore further.",
  ],
]
`;

exports[`test success message test matrix 'rust' 'vanilla' 'js' false 1`] = `
[
  [
    "
[32m======================================================[39m
‚úÖ  Success! Created 'my_project_name'
   with a smart contract in [1mRust[22m and a frontend template.
ü¶Ä If you are new to Rust please visit [1m[32mhttps://www.rust-lang.org [39m[22m

  [1m[43m[30mYour next steps[39m[49m[22m:
   - [7mNavigate to your project[27m:
         [34mcd [39m[34m[1mmy_project_name[22m[39m
   - [7mInstall all dependencies[27m
         [34mnpm [39m[34m[1minstall[22m[39m
   - [7mBuild your contract[27m:
         [34mnpm [39m[34m[1mrun build[22m[39m
   - [7mTest your contract[27m in NEAR SandBox:
         [34mnpm [39m[34m[1mtest[22m[39m
   - [7mDeploy your contract[27m to NEAR TestNet with a temporary dev account:
         [34mnpm [39m[34m[1mrun deploy[22m[39m
   - [7mStart your frontend[27m:
         [34mnpm [39m[34m[1mstart[22m[39m

üß† Read [1m[92mREADME.md[39m[22m to explore further.",
  ],
]
`;

exports[`test success message test matrix 'rust' 'vanilla' 'js' true 1`] = `
[
  [
    "
[32m======================================================[39m
‚úÖ  Success! Created 'my_project_name'
   with a smart contract in [1mRust[22m and a frontend template.
ü¶Ä If you are new to Rust please visit [1m[32mhttps://www.rust-lang.org [39m[22m

  [1m[43m[30mYour next steps[39m[49m[22m:
   - [7mNavigate to your project[27m:
         [34mcd [39m[34m[1mmy_project_name[22m[39m
   Then:
   - [7mBuild your contract[27m:
         [34mnpm [39m[34m[1mrun build[22m[39m
   - [7mTest your contract[27m in NEAR SandBox:
         [34mnpm [39m[34m[1mtest[22m[39m
   - [7mDeploy your contract[27m to NEAR TestNet with a temporary dev account:
         [34mnpm [39m[34m[1mrun deploy[22m[39m
   - [7mStart your frontend[27m:
         [34mnpm [39m[34m[1mstart[22m[39m

üß† Read [1m[92mREADME.md[39m[22m to explore further.",
  ],
]
`;

exports[`test success message test matrix 'rust' 'vanilla' 'rust' false 1`] = `
[
  [
    "
[32m======================================================[39m
‚úÖ  Success! Created 'my_project_name'
   with a smart contract in [1mRust[22m and a frontend template.
ü¶Ä If you are new to Rust please visit [1m[32mhttps://www.rust-lang.org [39m[22m

  [1m[43m[30mYour next steps[39m[49m[22m:
   - [7mNavigate to your project[27m:
         [34mcd [39m[34m[1mmy_project_name[22m[39m
   - [7mInstall all dependencies[27m
         [34mnpm [39m[34m[1minstall[22m[39m
   - [7mBuild your contract[27m:
         [34mnpm [39m[34m[1mrun build[22m[39m
   - [7mTest your contract[27m in NEAR SandBox:
         [34mnpm [39m[34m[1mtest[22m[39m
   - [7mDeploy your contract[27m to NEAR TestNet with a temporary dev account:
         [34mnpm [39m[34m[1mrun deploy[22m[39m
   - [7mStart your frontend[27m:
         [34mnpm [39m[34m[1mstart[22m[39m

üß† Read [1m[92mREADME.md[39m[22m to explore further.",
  ],
]
`;

exports[`test success message test matrix 'rust' 'vanilla' 'rust' true 1`] = `
[
  [
    "
[32m======================================================[39m
‚úÖ  Success! Created 'my_project_name'
   with a smart contract in [1mRust[22m and a frontend template.
ü¶Ä If you are new to Rust please visit [1m[32mhttps://www.rust-lang.org [39m[22m

  [1m[43m[30mYour next steps[39m[49m[22m:
   - [7mNavigate to your project[27m:
         [34mcd [39m[34m[1mmy_project_name[22m[39m
   Then:
   - [7mBuild your contract[27m:
         [34mnpm [39m[34m[1mrun build[22m[39m
   - [7mTest your contract[27m in NEAR SandBox:
         [34mnpm [39m[34m[1mtest[22m[39m
   - [7mDeploy your contract[27m to NEAR TestNet with a temporary dev account:
         [34mnpm [39m[34m[1mrun deploy[22m[39m
   - [7mStart your frontend[27m:
         [34mnpm [39m[34m[1mstart[22m[39m

üß† Read [1m[92mREADME.md[39m[22m to explore further.",
  ],
]
`;
